src/tests/stats/test_median.fypp

```diff
@@ -0,0 +1,196 @@
+                 '${k1}$ median(d2, 1)')
+    call check( sum(abs(median(d2, 2) - [3.5_dp, 1.5_dp, 3._dp])) < dptol,       &
+                 '${k1}$ median(d2, 2)')
+
+    !median_mask_all
+    call check( ieee_is_nan(median(d0, d0 > 0)), '${k1}$ median(d0, d0 > 0)' )
+    call check( ieee_is_nan(median(d1, d1 > huge(d1))), '${k1}$ median(d1, d1 > huge(d1))' )
+    call check( ieee_is_nan(median(d2, d2 > huge(d2))), '${k1}$ median(d2, d2 > huge(d2))' )
+    call check( ieee_is_nan(median(d3, d3 > huge(d3))), '${k1}$ median(d3, d3 > huge(d3))' )
+    call check( (median(d1, d1 > 0) - 7._dp) < dptol, '${k1}$ median(d1, d1 > 0)' ) 
+    call check( (median(d2, d2 > 0) - 7._dp) < dptol, '${k1}$ median(d2, d2 > 0)' ) 
+    call check( (median(d3, d3 > 0) - 7._dp) < dptol, '${k1}$ median(d3, d3 > 0)' ) 
+
+    !median mask
+    call check( ieee_is_nan(median(d0, 1, d0 > 0)), '${k1}$ median(d0, 1, d0 > 0)' )
+    call check( ieee_is_nan(median(d1, 1, d1 > huge(d1))), '${k1}$ median(d1, 1, d1 > huge(d1))' )
+    call check( any(ieee_is_nan(median(d2, 1, d2 > huge(d2)))), '${k1}$ median(d2, 1, d2 > huge(d2))' )
+    call check( any(ieee_is_nan(median(d3, 1, d3 > huge(d3)))), '${k1}$ median(d3, 1, d3 > huge(d3))' )
+    call check( (median(d1, 1, d1 > 0) - 7._dp) < dptol, '${k1}$ median(d1, 1, d1 >0)') 
+    call check( sum(abs( (median(d2, 1, d2 > 0) - [ 6._dp, 6._dp, 8._dp, 10.5_dp] )  )) &
+                 < dptol, '${k1}$ median(d2, 1, d2 > 0 )') 
+    call check( sum(abs( (median(d2, 2, d2 > 0) - [ 8.5_dp, 2._dp, 14.5_dp] )  ))       &
+                 < dptol, '${k1}$ median(d2, 2, d2 > 0 )') 
+    call check( any(ieee_is_nan(median(d3, 1, d3 > 0))), '${k1}$ median(d3, 1, d3 > 0)')
+
+    !odd number
+    d1 = [d1, 20_${k1}$]
+    call check(mod(size(d1), 2) == 1, 'test_median_${k1}$: should be an odd number')
+
+    call check( abs(median(d1) - 2._dp) < dptol, '${k1}$ median(d1), odd')
+
+    d2 = reshape(d1, [3, 5], pad = [0_${k1}$])
+    call check(mod(size(d2), 2) == 1, 'test_median_${k1}$: should be an odd number')
+    call check( abs(median(d2) - 1._dp) < dptol, '${k1}$ median(d2), odd')
+
+    d3 = reshape(d1, [1, 3, 5], pad = [0_${k1}$])
+    call check(mod(size(d3), 2) == 1, 'test_median_${k1}$: should be an odd number')
+    call check( abs(median(d3) - 1._dp) < dptol, '${k1}$ median(d3), odd')
+
+    call check( abs(median(d1, 1) - 2._dp) < dptol, '${k1}$ median(d1, 1)')
+    call check( sum(abs(median(d2, 1) - [2._dp, -4._dp, 7._dp, 1._dp, 0._dp])) < dptol, &
+                 '${k1}$ median(d2, 1), odd')
+    call check( sum(abs(median(d2, 2) - [7._dp, 1._dp, 0._dp])) < dptol,                &
+                 '${k1}$ median(d2, 2), odd')
+
+end subroutine
+#:endfor
+
+#:for k1,t1 in REAL_KINDS_TYPES
+subroutine test_median_${k1}$()
+    ${t1}$, allocatable :: d0(:), d1(:), d2(:,:), d3(:,:,:)
+
+    allocate(d0(0))
+    call check(size(d0) == 0, 'test_median_${k1}$: should be of size 0')
```

Comment:
I don't understand the purpose of this test.

Reply:
Yes, it is just to make sure the setup of d0 is correct. I will add a comment.